src/main/java/hudson/plugins/gradle/injection/MavenBuildScanInjection.java

```diff
@@ -136,18 +86,12 @@ private void inject(Node node, FilePath nodeRootPath) {
                 systemProperties.add(new SystemProperty(GRADLE_ENTERPRISE_ALLOW_UNTRUSTED_SERVER_PROPERTY_KEY, "true"));
             }
 
-            MAVEN_OPTS_SETTER.append(node, systemProperties);
+            EnvUtil.setEnvVar(node, JENKINSGRADLEPLUGIN_MAVEN_OPTS_PREPARED, MAVEN_OPTS_HANDLER.merge(node, systemProperties));
```

Comment:
I don't think that `PREPARED` suffix brings any value here since we already have our prefix, which is enough for Maven to ignore it. Maybe just `JENKINSGRADLEPLUGIN_MAVEN_OPTS`?

Reply:
For MAVEN_OPTS yes, this suffix is redundant, but we already have JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH variable and need another one to distinguish between them. 
So I just kept the naming of these aligned.

Reply:
Why you can't reuse the same variable name? We do copying exactly for that reason, that the initially set variable is not available in the context where we need it, otherwise, nothing needs to be done.

Reply:
As discussed, we need both of these and I think it's clearer to just set 2 separate variables instead of one and do splitting afterwards